presto-parquet/src/test/java/com/facebook/presto/parquet/writer/TestParquetWriter.java

```diff
@@ -0,0 +1,111 @@
+
+    @BeforeMethod
+    public void setup()
+    {
+        temporaryDirectory = createTempDir();
+        parquetFile = new File(temporaryDirectory, randomUUID().toString());
+    }
+
+    @AfterMethod
+    public void tearDown()
+            throws IOException
+    {
+        deleteRecursively(temporaryDirectory.toPath(), ALLOW_INSECURE);
+    }
+
+    @Test
+    public void testRowGroupFlushInterleavedColumnWriterFallbacks()
+    {
+        List<Type> types = ImmutableList.of(BIGINT, INTEGER, VARCHAR, BOOLEAN);
+        List<String> names = ImmutableList.of("col_1", "col_2", "col_3", "col_4");
+        ParquetWriterOptions parquetWriterOptions = ParquetWriterOptions.builder()
+                .setMaxPageSize(DataSize.succinctBytes(1000))
+                .setMaxBlockSize(DataSize.succinctBytes(15000))
+                .setMaxDictionaryPageSize(DataSize.succinctBytes(1000))
+                .build();
+        try (ParquetWriter parquetWriter = createParquetWriter(parquetFile, types, names, parquetWriterOptions, CompressionCodecName.UNCOMPRESSED)) {
+            Random rand = new Random();
+            for (int pageIdx = 0; pageIdx < 10; pageIdx++) {
+                int pageRowCount = 100;
+                PageBuilder pageBuilder = new PageBuilder(pageRowCount, types);
+                for (int rowIdx = 0; rowIdx < pageRowCount; rowIdx++) {
+                    // maintain col_1's dictionary size approximately half of raw data
+                    BIGINT.writeLong(pageBuilder.getBlockBuilder(0), pageIdx * 100 + rand.nextInt(50));
+                    INTEGER.writeLong(pageBuilder.getBlockBuilder(1), rand.nextInt(100000000));
+                    VARCHAR.writeString(pageBuilder.getBlockBuilder(2), UUID.randomUUID().toString());
+                    BOOLEAN.writeBoolean(pageBuilder.getBlockBuilder(3), rand.nextBoolean());
+                    pageBuilder.declarePosition();
+                }
+                parquetWriter.write(pageBuilder.build());
+            }
+        }
+        catch (Exception e) {
+            fail("Should not fail!");
+        }
+    }
+
+    public static ParquetWriter createParquetWriter(File outputFile, List<Type> types, List<String> columnNames,
+                                                    ParquetWriterOptions parquetWriterOptions, CompressionCodecName compressionCodecName)
+            throws Exception
+    {
+        checkArgument(types.size() == columnNames.size());
+        ParquetSchemaConverter schemaConverter = new ParquetSchemaConverter(
+                types,
+                columnNames);
+        ParquetWriter writer = new ParquetWriter(
+                new FileOutputStream(outputFile),
+                schemaConverter.getMessageType(),
+                schemaConverter.getPrimitiveTypes(),
+                columnNames,
+                types,
+                parquetWriterOptions,
+                compressionCodecName.getHadoopCompressionCodecClassName());
+        return writer;
```

Comment:
You can simply return to line 101.

Reply:
Thanks a lot for pointing it out, I have fixed it.